crates/e2e/tests/e2e/unwrap_cost_optimization.rs

```diff
@@ -0,0 +1,388 @@
+const TRADER_BUY_ETH_B_PK: [u8; 32] = [2; 32];
+
+const ORDER_PLACEMENT_ENDPOINT: &str = "/api/v1/orders/";
+const FEE_ENDPOINT: &str = "/api/v1/fee/";
+
+#[tokio::test]
+#[ignore]
+async fn local_node_optimize_internal_unwraps() {
+    crate::local_node::test(optimize_internal_unwraps).await;
+}
+
+async fn fund_accounts(
+    web3: &Web3,
+    contracts: &Contracts,
+) -> (Account, Account, Account, contracts::ERC20Mintable) {
+    let accounts: Vec<Address> = web3.eth().accounts().await.expect("get accounts failed");
+    let solver = Account::Local(accounts[0], None);
+    let weth_donor = Account::Local(accounts[1], None);
+    let trader_a = Account::Offline(PrivateKey::from_raw(TRADER_BUY_ETH_A_PK).unwrap(), None);
+    let trader_b = Account::Offline(PrivateKey::from_raw(TRADER_BUY_ETH_B_PK).unwrap(), None);
+    let token = deploy_mintable_token(web3).await;
+
+    for trader in [&trader_a, &trader_b] {
+        // reset ETH and token balance in case this is not the first time the account has been used
+        let weth_balance = eth_balance(web3, trader.address()).await;
+        tx_value!(trader, weth_balance, contracts.weth.deposit());
+        let token_balance = token.balance_of(trader.address()).call().await.unwrap();
+        token
+            .transfer(weth_donor.address(), token_balance)
+            .call()
+            .await
+            .unwrap();
+
+        // fund traders with 51 token and set allowance for settlement contract
+        tx!(solver, token.mint(trader.address(), to_wei(51)));
+        tx!(trader, token.approve(contracts.allowance, to_wei(51)));
+    }
+
+    // fund settlement contract with 100 WETH
+    tx_value!(weth_donor, to_wei(100), contracts.weth.deposit());
+    contracts
+        .weth
+        .transfer(contracts.gp_settlement.address(), to_wei(100))
+        .from(weth_donor)
+        .send()
+        .await
+        .unwrap();
+
+    // Create and fund Uniswap pool
+    // 100K WETH <=> 100K token_a
+    tx!(solver, token.mint(solver.address(), to_wei(100_000)));
+    tx_value!(solver, to_wei(100_000), contracts.weth.deposit());
+    tx!(
+        solver,
+        contracts
+            .uniswap_factory
+            .create_pair(token.address(), contracts.weth.address())
+    );
+    tx!(
+        solver,
+        token.approve(contracts.uniswap_router.address(), to_wei(100_000))
+    );
+    tx!(
+        solver,
+        contracts
+            .weth
+            .approve(contracts.uniswap_router.address(), to_wei(100_000))
+    );
+    tx!(
+        solver,
+        contracts.uniswap_router.add_liquidity(
+            token.address(),
+            contracts.weth.address(),
+            to_wei(100_000),
+            to_wei(100_000),
+            0_u64.into(),
+            0_u64.into(),
+            solver.address(),
+            U256::max_value(),
+        )
+    );
+
+    (solver, trader_a, trader_b, token)
+}
+
+async fn set_up_driver_and_services(
```

Comment:
Could we avoid calling `OrderbookServices::new` twice for one test by separating this function to two functions? Is it necessary to have two `OrderbookServices` objects in one test?